googlebigtable2/src/main/java/site/ycsb/db/GoogleBigtable2Client.java

```diff
@@ -0,0 +1,420 @@
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package site.ycsb.db;
+
+import com.google.api.gax.batching.Batcher;
+import com.google.api.gax.batching.BatchingException;
+import com.google.api.gax.batching.BatchingSettings;
+import com.google.cloud.bigtable.data.v2.BigtableDataClient;
+import com.google.cloud.bigtable.data.v2.BigtableDataSettings;
+import com.google.cloud.bigtable.data.v2.models.Filters;
+import com.google.cloud.bigtable.data.v2.models.MutationApi;
+import com.google.cloud.bigtable.data.v2.models.Query;
+import com.google.cloud.bigtable.data.v2.models.Range;
+import com.google.cloud.bigtable.data.v2.models.Row;
+import com.google.cloud.bigtable.data.v2.models.RowCell;
+import com.google.cloud.bigtable.data.v2.models.RowMutation;
+import com.google.cloud.bigtable.data.v2.models.RowMutationEntry;
+import com.google.cloud.bigtable.data.v2.models.TableId;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import site.ycsb.ByteIterator;
+import site.ycsb.DBException;
+import site.ycsb.Status;
+
+/**
+ * Idiomatic Java client for Google Bigtable Proto client for YCSB framework.
+ */
+public class GoogleBigtable2Client extends site.ycsb.DB {
+
+  /**
+   * Property names for the CLI.
+   */
+  private static final String PROP_PREFIX = "googlebigtable2";
+
+  private static final String DEBUG_KEY = "debug";
+  private static final String ENDPOINT_KEY = PROP_PREFIX + ".data-endpoint";
+  private static final String PROJECT_KEY = PROP_PREFIX + ".project";
+  private static final String INSTANCE_KEY = PROP_PREFIX + ".instance";
+  private static final String APP_PROFILE_ID_KEY = PROP_PREFIX + ".app-profile";
+  private static final String FAMILY_KEY = PROP_PREFIX + ".family";
+
+  private static final String MAX_OUTSTANDING_BYTES = PROP_PREFIX + ".max-outstanding-bytes";
+  private static final String CLIENT_SIDE_BUFFERING = PROP_PREFIX + ".use-batching";
+
+  /**
+   * Print debug information to standard out.
+   */
+  private static boolean debug = false;
+
+  /**
+   * Tracks running thread counts so we know when to close the session.
+   */
+  private static int clientRefCount = 0;
```

Comment:
should this be atomicInteger or volatile?

Reply:
its only ever updated in synchronized methods